src/main/java/eu/arrowhead/blacklist/jpa/service/EntryDbService.java

```diff
@@ -0,0 +1,227 @@
+					&& Utilities.isEmpty(revokers)
+					&& Utilities.isEmpty(reason)
+					&& alivesAt == null) {
+				entries = entryRepo.findAll(pagination);
+			}
+
+			// With filters
+			if (entries == null) {
+				synchronized (LOCK) {
+					final List<Long> matchingIDs = new ArrayList<>();
+					final List<Entry> toFilter = Utilities.isEmpty(systemNames) ? entryRepo.findAll() : entryRepo.findAllBySystemNameIn(systemNames);
+					for (final Entry entry : toFilter) {
+						if (mode != null && !modeMatch(mode, entry.getActive())) {
+							continue;
+						}
+						if (!Utilities.isEmpty(issuers) && !issuers.contains(entry.getCreatedBy())) {
+							continue;
+						}
+						if (!Utilities.isEmpty(revokers) && (entry.getRevokedBy() == null || !revokers.contains(entry.getRevokedBy()))) {
+							continue;
+						}
+						// reason match is case insensitive
+						if (!Utilities.isEmpty(reason) && !entry.getReason().toLowerCase().contains(reason.toLowerCase())) {
+							continue;
+						}
+						if (entry.getExpiresAt() != null && alivesAt != null && !(alivesAt.isBefore(entry.getExpiresAt()) || alivesAt.isEqual(entry.getExpiresAt()))) {
+							continue;
+						}
+
+						matchingIDs.add(entry.getId());
+					}
+					entries = entryRepo.findAllByIdIn(matchingIDs, pagination);
+				}
+			}
+			return entries;
+		} catch (final Exception ex) {
+			logger.error(ex.getMessage());
+			logger.debug(ex);
+			throw new InternalServerError("Database operation error");
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public void unsetActiveByNameList(final List<String> names, final String revokerName, final String origin) {
+		logger.debug("unsetActiveByNameList started...");
+		Assert.isTrue(!Utilities.isEmpty(names), "System name list is missing or empty");
+
+		try {
+			synchronized (LOCK) {
+				final List<Entry> toInactivate = entryRepo.findAllBySystemNameIn(names).stream().filter(Entry::getActive).collect(Collectors.toList());
+				for (final Entry entry : toInactivate) {
+					entry.setActive(false);
+					entry.setRevokedBy(revokerName);
+				}
+				entryRepo.saveAllAndFlush(toInactivate);
+			}
+		} catch (final Exception ex) {
+			logger.error(ex.getMessage());
+			logger.debug(ex);
+			throw new InternalServerError("Database operation error");
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	// Returns true, if there is a record with the given system name, the active flag is set, and the expiration date is in the future.
+	public boolean isActiveEntryForName(final String systemName, final String origin) {
+		logger.debug("isActiveEntryForName started, name: {}", systemName);
+		Assert.isTrue(!Utilities.isEmpty(systemName), "System name is missing or empty");
+
+		try {
+			synchronized (LOCK) {
+
+				// finding the matching system name
+				final List<Entry> entries = entryRepo.findAllBySystemName(systemName);
```

Comment:
instead of this repo calling you should define findAllBySystemNameAndActive() and use that